### PR TITLE
Refactor Accounts API endpoints to follow REST standards - Closes #903, #933

### LINKED ISSUE #903: Refactor Accounts API endpoints to follow REST standards

### PR DESCRIPTION
**Pull request by SargeKhan**
Closes #903, Closes #933

### CODE REVIEW

File: helpers/orderBy.js [hunk 1]
```diff
@@ -82,11 +82,15 @@ function OrderBy (orderBy, options) {
  * to format accepted by "json-sql" library: {field: 1}.
  * Ascending sort method number equivalent is 1.
  * Descending sort method number equivalent is -1.
- * @param {string} sortQuery
+ * If only field is sepcified in sortQuery, sortOrder will be ascending.
```
Comment (Isabello) [inline helpers/orderBy.js]: specified

File: logic/account.js [hunk 2]
```diff
@@ -7,9 +7,12 @@ var jsonSql = require('json-sql')();
 jsonSql.setDialect('postgresql');
 var constants = require('../helpers/constants.js');
 var slots = require('../helpers/slots.js');
+var orderBy = require('../helpers/orderBy.js');
+var BlockReward = require('../logic/blockReward.js');
```
Comment (Isabello) [inline logic/account.js]: inconsistent casing.

should be:
```
blockReward
bignum
```

File: logic/account.js [hunk 3]
```diff
@@ -609,12 +676,49 @@ Account.prototype.getAll = function (filter, fields, cb) {
...
+			}
+		}
 	});
 
+	var self = this;
+
 	this.scope.db.query(sql.query, sql.values).then(function (rows) {
+		var lastBlock = modules.blocks.lastBlock.get();
+		// If the last block height is undefined, it means it's a genesis block with height = 1
+		// look for aconstant for total suppply
```
Comment (Isabello) [inline logic/account.js]: a constant

Comment (Isabello) [reply logic/account.js]: supply

File: modules/accounts.js [hunk 4]
```diff
@@ -2,12 +2,16 @@
 
 var bignum = require('../helpers/bignum.js');
 var BlockReward = require('../logic/blockReward.js');
+var slots = require('../helpers/slots.js');
 var constants = require('../helpers/constants.js');
 var crypto = require('crypto');
+var apiCodes = require('../helpers/apiCodes.js');
+var ApiError = require('../helpers/apiError.js');
```
Comment (Isabello) [inline modules/accounts.js]: Casing

File: modules/accounts.js [hunk 5]
```diff
@@ -2,12 +2,16 @@
...
 var extend = require('extend');
 var schema = require('../schema/accounts.js');
 var transactionTypes = require('../helpers/transactionTypes.js');
 var Vote = require('../logic/vote.js');
+var _ = require('lodash');
```
Comment (Isabello) [inline modules/accounts.js]: Why do we need to use lodash? Missing native features in ES5?

File: schema/accounts.js [hunk 6]
```diff
@@ -1,69 +1,39 @@
...
-	top: {
-		id: 'accounts.top',
-		type: 'object',
-		properties: {
 			limit: {
 				type: 'integer',
-				minimum: 0,
-				maximum: 100
+				minimum: 1,
+				maximum: constants.activeDelegates
```
Comment (Isabello) [inline schema/accounts.js]: Good!

File: test/unit/modules/accounts.js [hunk 7]
```diff
@@ -84,6 +84,7 @@ describe('account', function () {
 			expect(accounts.generateAddressByPublicKey(validAccount.publicKey)).to.equal(validAccount.address);
 		});
 
+		// ToDo: Design a throwable test
```
Comment (Isabello) [inline test/unit/modules/accounts.js]: TODO

File: logic/account.js (hunk 2 above)
Comment (SargeKhan) [reply logic/account.js]: I think we aren't following best practice in the other parts of the code. The best practice is that constructor functions should start with a capital letter. `BlockReward` and `Bignum` are two constructor functions.

File: logic/account.js (hunk 3 above)
Comment (SargeKhan) [reply logic/account.js]: thanks!

File: modules/accounts.js (hunk 5 above)
Comment (SargeKhan) [reply modules/accounts.js]: I think you are right, we can do without lodash in this file.

File: test/unit/modules/accounts.js (hunk 7 above)
Comment (SargeKhan) [reply test/unit/modules/accounts.js]: thanks

File: logic/account.js (hunk 2 above)
Comment (Isabello) [reply logic/account.js]: Ok

**Pull request merged**